ssdp.md

```diff
@@ -0,0 +1,135 @@
+* The controller starts "Monitor Display Availability" algorithm by sending a SSDP search message with the service type `urn:example-org:service:osp:1` and waits for responses from devices running a presentation receiver service. The controller should wait for SSDP alive/byebye messages on the multicast address to keep the list of available displays up-to-date (e.g. when a new display in added or an existing one is removed).  
+* Each display available on the network and runs a presentation receiver service replies to the search request with a SSDP message similar to the alive message.
+* The controller reads the LOCATION header for each available display and makes a HTTP GET request to get the device description XML. 
+* The controller parses each device description XML, reads the friendly Name of the display and checks if the display can open one of the URLs passed as input to the PresentationRequest constuctor. If yes, the device will be added to the list of available displays.
+* Open questions:
+    * how to send the endpoint of the receiver service: one solution is to use a HTTP header parameter in the response of the HTTP GET request for device description. DIAL uses this solution to send the endpoint of the DIAL server in the `Application-URL` HTTP response header. Another solution is to extend the XML device description with a new element to define the endpoint.
+    * How to check if the display can open a certain URL or not. One solution is to extend the XML device description with new elements to allow a display to express its capabilities and the controller can do the check. Another possible solution is to ask the receiver service using the provided endpoint by sending the presentation URLs. 
+
+**Method 2**: This method uses only the SSDP messages without requesting the device description XML. Idea is to send the presentation URLs in a new header of the SSDP search message. Only receivers that can open at least one of the URLs response to the search request. The search response contains also a new header for device friendly name and another header for service endpoint (section 1.1.3 of the UPnP device architecture document allows to use vendor specific headers). The search response can still send the device description URL in the Location header to stay compatible with UPnP but controller devices don’t need to use it since all necessary information are send directly in the SSDP search response. This method is more efficient since no additional HTTP calls and XML parsing is needed. Below are the steps that illustrate the idea of this method:
+
+* New display with presentation receiver service that appears on the network advertises the following SSDP alive message. The new header `FRIENDLY-NAME.example.org` is used for friendly name and `PRESENTATION-ENDPOINT.example.org` for endpoint of the receiver service. The value `urn:example-org:service:osp:1` of header `NT` is used for presentation receiver service.
```

Comment:
UPDATE: DIAL introduces in version 2.0 a new SSDP header `WAKEUP` in SEARCH responses from presentation devices that support WoL. @mwatson2 do yo have more details on this? is `WAKEUP` introduced in DIAL oder somewhere else?